File: Problem Set 6/sentimental-readability/readability.py
# TODO

from cs50 import get_string

text = get_string("Text: ")

# letter count we will count the alphabetic characters
lettercount = 0
# word count we will count the spaces between words but start the count at 1 as the first word won't have a space
wordcount = 1
# sentence count will count the specific characters ., !, ?
sentencecount = text.count(".") + text.count("!") + text.count("?")

for character in text:
    if character.isalpha():
        lettercount += 1
    elif character.isspace():
        wordcount += 1

L = lettercount / wordcount * 100
S = sentencecount / wordcount * 100

ColemanIndex = 0.0588 * L - 0.296 * S - 15.8
# need to not truncate the value of the Coleman index but then round to nearest integer
grade = round(ColemanIndex, 0)

if grade > 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print("Grade %2d" % grade)

File: Problem Set 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{

    string A = get_string("Text: ");

    int letters = count_letters(A);
    int words = count_words(A);
    int sentences = count_sentences(A);

    float L = ((float) letters / (float) words) * 100;
    float S = ((float) sentences / (float) words) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}


int count_letters(string text)
{
    int n = strlen(text);
    int i;
    int count = 0;

    for (i = 0; i < n; i++)
    {
        if (islower(text[i]) > 0 || isupper(text[i]) > 0)
        {
            count++;
        }
        else
        {
            count += 0;
        }
    }
    return count;
}


int count_words(string text)
{
    int n = strlen(text);
    int i;
    int count = 1;

    for (i = 0; i < n; i++)
    {
        if ((int) text[i] == 32)
        {
            count++;
        }
        else
        {
            count += 0;
        }
    }
    return count;
}


int count_sentences(string text)
{
    int n = strlen(text);
    int i;
    int count = 0;

    for (i = 0; i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
        else
        {
            count += 0;
        }
    }
    return count;
}